modules/rostests/apitests/compiler/floatconv.c

```diff
@@ -0,0 +1,272 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Test for floating point conversion
+ * COPYRIGHT:   Copyright 2020 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#include <apitest.h>
+
+/*
+ * Note: There are 2 behaviors for float to unsigned long conversion:
+ * 1. The old behavior, which exists only on x86 and CL versions up to somewhere
+ *    between 19.40.33811 and 19.41.33923:
+ *     - If a negative float is cast to an unsigned long, the result is ULONG_MAX
+ *     - If a float is cast to an unsigned long and the value is larger than ULONG_MAX,
+ *           the result is ULONG_MAX
+ * 2. The new behavior, which exists for x64 and on x86 versions after the ones mentioned above:
+ *     - If a neative float is cast to an unsigned long, the result is the same as casting to a long
+ *     - If a float is cast to an unsigned long and the value is larger than ULONG_MAX,
+ *
+ * In the old version, the float to unsigned conversion was inlined and used _ftoul2 and
+ * a comparison against 0x43e0000000000000 (9223372036854775808.0) to check for overflow.
+ * In the new version, the float to unsigned conversion is done by a call to _ftoul2_legacy,
+ * which checks for overflow by comparing against 0x5f800000 (18446744073709551616.0) and
+ * then forwards the call to either _ftol2 or _ftoul2.
+ */
+
+#ifdef _M_IX86
+#ifdef __REACTOS__
+#define OLD_BEHAVIOR1
```

Comment:
`OLD_BEHAVIOR1` seems unused...

Reply:
This is a draft and you haven't been requested to review this!